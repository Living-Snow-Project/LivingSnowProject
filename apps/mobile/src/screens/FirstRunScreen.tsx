import React from "react";
import { Box, Button, HStack, VStack, Text } from "native-base";
import { SnowIcon, UserIdentityInput } from "../components";
import { setAppSettings } from "../../AppSettings";
import { FirstRunScreenNavigationProp } from "../navigation/Routes";
import { Labels } from "../constants";

type FirstRunScreenProps = {
  navigation: FirstRunScreenNavigationProp;
};

export function FirstRunScreen({ navigation }: FirstRunScreenProps) {
  return (
    <Box px={3} mt={2}>
      <HStack alignItems="center" justifyContent="space-between">
        <SnowIcon />
        <SnowIcon />
        <VStack>
          <Text fontWeight="bold" fontSize="xl" color="pink.600">
            {Labels.LivingSnowProject}
          </Text>
          <Text color="pink.500">{Labels.Slogan}</Text>
        </VStack>
        <SnowIcon />
        <SnowIcon />
      </HStack>

      <Text my={2}>{Labels.FirstRunScreen.Usage}</Text>

      <UserIdentityInput />

      <Box alignItems="center">
        <Button
          mt={5}
          width={200}
          onPress={() => {
            setAppSettings((prev) => ({ ...prev, showFirstRun: false }));
            navigation.navigate("Timeline");
          }}
        >
          {Labels.FirstRunScreen.StartReporting}
        </Button>
      </Box>
    </Box>
  );
}
