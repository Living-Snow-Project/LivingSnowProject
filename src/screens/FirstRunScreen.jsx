import React, { useContext } from "react";
import Touchable from "react-native-platform-touchable";
import { Platform, StyleSheet, Text, View } from "react-native";
import { StockIcon } from "../components/TabBarIcon";
import { AppSettingsContext } from "../../AppSettings";
import UserIdentityInput from "../components/forms/UserIdentityInput";
import Routes from "../navigation/Routes";

const styles = StyleSheet.create({
  ftreContainer: {
    flex: 1,
    marginTop: 30,
    paddingHorizontal: 10,
  },
  welcomeContainer: {
    flexDirection: "row",
    justifyContent: "center",
  },
  exitContainer: {
    flex: 2,
    justifyContent: "flex-start",
    alignItems: "center",
    marginTop: 30,
  },
  exitButtonContainer: {
    backgroundColor: "lightpink",
    width: 200,
    height: 40,
    borderRadius: 5,
    justifyContent: "center",
  },
  exitButtonText: {
    fontSize: 20,
    fontWeight: "bold",
    textAlign: "center",
  },
  welcomeText: {
    fontSize: 25,
    marginTop: 1,
    fontWeight: "bold",
    textAlign: "center",
  },
  descriptionText: {
    fontSize: 12,
    marginTop: 20,
    marginBottom: 20,
    marginLeft: 20,
    marginRight: 20,
    color: "red",
    textAlign: "center",
  },
  snowLeft: { marginRight: 30 },
  snowRight: { marginLeft: 30 },
});

export default function FirstRunScreen({ navigation }) {
  const { updateAppSettings } = useContext(AppSettingsContext);

  return (
    <View style={styles.ftreContainer}>
      <View style={styles.welcomeContainer}>
        <StockIcon
          style={styles.snowLeft}
          name={Platform.OS === "ios" ? "ios-snow" : "md-snow"}
        />
        <Text style={styles.welcomeText}>Living Snow Project</Text>
        <StockIcon
          style={styles.snowRight}
          name={Platform.OS === "ios" ? "ios-snow" : "md-snow"}
        />
      </View>

      <Text style={styles.descriptionText}>
        {"Enter your name and the organization you are associated with, or skip and remain anonymous." +
          " You can change these at any time in the Settings tab."}
      </Text>

      <UserIdentityInput />

      <View style={styles.exitContainer}>
        <Touchable
          onPress={() => {
            updateAppSettings({ showFirstRun: false });
            navigation.navigate(Routes.TimelineScreen);
          }}
        >
          <View style={styles.exitButtonContainer}>
            <Text style={styles.exitButtonText}>Let&apos;s get started!</Text>
          </View>
        </Touchable>
      </View>
    </View>
  );
}
