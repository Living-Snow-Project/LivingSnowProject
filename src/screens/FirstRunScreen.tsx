import React from "react";
import { Platform, ScrollView, StyleSheet, Text, View } from "react-native";
import PressableOpacity from "../components/PressableOpacity";
import { StockIcon } from "../components/Icons";
import { getAppSettings, setAppSettings } from "../../AppSettings";
import UserIdentityInput from "../components/forms/UserIdentityInput";
import KeyboardShift from "../components/KeyboardShift";

const styles = StyleSheet.create({
  ftreContainer: {
    flex: 1,
    marginTop: 30,
    paddingHorizontal: 10,
  },
  welcomeContainer: {
    flexDirection: "row",
    justifyContent: "center",
  },
  exitContainer: {
    flex: 2,
    justifyContent: "flex-start",
    alignItems: "center",
    marginTop: 30,
  },
  exitButtonContainer: {
    backgroundColor: "lightpink",
    width: 200,
    height: 40,
    borderRadius: 5,
    justifyContent: "center",
  },
  exitButtonText: {
    fontSize: 20,
    fontWeight: "bold",
    textAlign: "center",
  },
  welcomeText: {
    fontSize: 25,
    marginTop: 1,
    fontWeight: "bold",
    textAlign: "center",
  },
  descriptionText: {
    fontSize: 12,
    marginTop: 20,
    marginBottom: 20,
    marginLeft: 20,
    marginRight: 20,
    color: "red",
    textAlign: "center",
  },
  snowLeft: { marginRight: 30 },
  snowRight: { marginLeft: 30 },
});

export default function FirstRunScreen({ navigation }) {
  return (
    <KeyboardShift>
      {() => (
        <ScrollView style={styles.ftreContainer}>
          <View style={styles.welcomeContainer}>
            <StockIcon
              style={styles.snowLeft}
              name={Platform.OS === "ios" ? "ios-snow" : "md-snow"}
            />
            <Text style={styles.welcomeText}>Living Snow Project</Text>
            <StockIcon
              style={styles.snowRight}
              name={Platform.OS === "ios" ? "ios-snow" : "md-snow"}
            />
          </View>

          <Text style={styles.descriptionText}>
            {"Enter your name and the organization you are associated with, or skip and remain anonymous." +
              " You can change these at any time in the Settings tab."}
          </Text>

          <UserIdentityInput />

          <View style={styles.exitContainer}>
            <PressableOpacity
              onPress={() => {
                setAppSettings({ ...getAppSettings(), showFirstRun: false });
                navigation.navigate("Timeline");
              }}
            >
              <View style={styles.exitButtonContainer}>
                <Text style={styles.exitButtonText}>
                  Let&apos;s get started!
                </Text>
              </View>
            </PressableOpacity>
          </View>
        </ScrollView>
      )}
    </KeyboardShift>
  );
}
