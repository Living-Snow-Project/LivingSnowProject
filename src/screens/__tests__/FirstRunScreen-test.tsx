import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import FirstRunScreen from "../FirstRunScreen";

test("validates Welcome Screen navigation", () => {
  const navigation = {
    navigate: jest.fn(),
  };

  const { getByText, toJSON } = render(
    <FirstRunScreen navigation={navigation} />
  );
  fireEvent.press(getByText("Let's get started!"));
  expect(navigation.navigate).toBeCalledTimes(1);
  expect(toJSON).toMatchSnapshot();
});
