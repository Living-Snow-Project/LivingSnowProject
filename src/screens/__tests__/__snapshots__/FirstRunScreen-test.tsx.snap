// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validates Welcome Screen navigation 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "transform": Array [
        Object {
          "translateY": 0,
        },
      ],
      "width": "100%",
    }
  }
>
  <RCTScrollView
    style={
      Object {
        "flex": 1,
        "marginTop": 30,
        "paddingHorizontal": 10,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text />
        <Text
          style={
            Object {
              "fontSize": 25,
              "fontWeight": "bold",
              "marginTop": 1,
              "textAlign": "center",
            }
          }
        >
          Living Snow Project
        </Text>
        <Text />
      </View>
      <Text
        style={
          Object {
            "color": "red",
            "fontSize": 12,
            "marginBottom": 20,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 20,
            "textAlign": "center",
          }
        }
      >
        Enter your name and the organization you are associated with, or skip and remain anonymous. You can change these at any time in the Settings tab.
      </Text>
      <Text
        style={
          Object {
            "fontSize": 15,
            "paddingLeft": 10,
            "paddingVertical": 2,
          }
        }
      >
        Name
      </Text>
      <TextInput
        allowFontScaling={true}
        maxLength={50}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Enter your name"
        rejectResponderTermination={true}
        returnKeyType="done"
        style={
          Object {
            "backgroundColor": "#efefef",
            "borderColor": "gray",
            "borderRadius": 4,
            "borderWidth": 1,
            "marginLeft": 10,
            "marginRight": 10,
            "paddingHorizontal": 15,
            "paddingVertical": 15,
          }
        }
        underlineColorAndroid="transparent"
      />
      <Text
        style={
          Object {
            "fontSize": 15,
            "paddingLeft": 10,
            "paddingVertical": 2,
          }
        }
      >
        Organization
      </Text>
      <TextInput
        allowFontScaling={true}
        maxLength={50}
        onChangeText={[Function]}
        placeholder="Enter organization you belong to (if any)"
        rejectResponderTermination={true}
        returnKeyType="done"
        style={
          Object {
            "backgroundColor": "#efefef",
            "borderColor": "gray",
            "borderRadius": 4,
            "borderWidth": 1,
            "marginLeft": 10,
            "marginRight": 10,
            "paddingHorizontal": 15,
            "paddingVertical": 15,
          }
        }
        underlineColorAndroid="transparent"
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 2,
            "justifyContent": "flex-start",
            "marginTop": 30,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {},
              Object {
                "opacity": 1,
              },
            ]
          }
          testID=""
        >
          <View
            style={
              Object {
                "backgroundColor": "lightpink",
                "borderRadius": 5,
                "height": 40,
                "justifyContent": "center",
                "width": 200,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Let's get started!
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
